docs/en/Building-Alluxio-From-Source.md

```diff
@@ -13,7 +13,7 @@ priority: 1
 
 This guide describes how to compile Alluxio from the beginning.
 
-The prerequisite for this guide is that you have [Java 8 or later](Java-Setup.html), [Maven 3.3.9 or later](Maven.html) installed.
+The prerequisite for this guide is that you must have [Java 8](Java-Setup.html) and [Maven 3.3.9 or later](Maven.html) installed.
```

Comment:
did you fail when using Java 9 to build from src?

Reply:
I did when I tried, but with the different versions being installed something could have interfered. I will try on java 9 again with a clean environment from a base ubuntu 16.04 image and confirm.

Reply:
thanks. in theory, the source code should build with Java 9 --- that's the goal. If it is not, I would create a JIRA to fix the build with Java 9  instead of fixing the doc.

Reply:
Here's what I found: the [Oracle JDK 9 isn't available any more](https://www.oracle.com/technetwork/java/javase/downloads/jdk9-downloads-3848520.html) and openjdk-9 isn't available in 18.04 repositories any more. openjdk-9 _is_ available in the 16.04 repositories, but when trying to build, I get an issue related to the version string (seems like a maven plugin issue?):


```
INFO] --- findbugs-maven-plugin:3.0.2:findbugs (findbugs) @ alluxio-core-protobuf ---
[INFO] Fork Value is true
     [java] Warning: Unknown version of Java
     [java] edu.umd.cs.findbugs.JavaVersionException: Could not parse Java version string: 9-internal
     [java]     at edu.umd.cs.findbugs.JavaVersion.<init>(JavaVersion.java:68)
     [java]     at edu.umd.cs.findbugs.JavaVersion.<clinit>(JavaVersion.java:44)
     [java]     at edu.umd.cs.findbugs.DetectorFactory.isEnabledForCurrentJRE(DetectorFactory.java:248)
     [java]     at edu.umd.cs.findbugs.FindBugs.isDetectorEnabled(FindBugs.java:256)
     [java]     at edu.umd.cs.findbugs.FindBugs2$3.choose(FindBugs2.java:906)
     [java]     at edu.umd.cs.findbugs.plan.ExecutionPlan.build(ExecutionPlan.java:179)
     [java]     at edu.umd.cs.findbugs.FindBugs2.createExecutionPlan(FindBugs2.java:935)
     [java]     at edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:241)
     [java]     at edu.umd.cs.findbugs.FindBugs.runMain(FindBugs.java:402)
     [java]     at edu.umd.cs.findbugs.FindBugs2.main(FindBugs2.java:1200)
     [java] The following classes needed for analysis were missing:
     [java]   java.lang.Object
     [java]   java.lang.Enum
     [java]   java.io.Serializable
     [java]   java.lang.Cloneable
     [java] Exception in thread "main" java.lang.RuntimeException: Could not instantiate edu.umd.cs.findbugs.detect.BuildObligationPolicyDatabase as Detector2
     [java]     at edu.umd.cs.findbugs.DetectorFactory$ReflectionDetectorCreator.createDetector2(DetectorFactory.java:94)
     [java]     at edu.umd.cs.findbugs.DetectorFactory.createDetector2(DetectorFactory.java:396)
     [java]     at edu.umd.cs.findbugs.plan.AnalysisPass.instantiateDetector2sInPass(AnalysisPass.java:132)
     [java]     at edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:993)
     [java]     at edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:283)
     [java]     at edu.umd.cs.findbugs.FindBugs.runMain(FindBugs.java:402)
     [java]     at edu.umd.cs.findbugs.FindBugs2.main(FindBugs2.java:1200)
     [java] Caused by: java.lang.reflect.InvocationTargetException
     [java]     at sun.reflect.NativeConstructorAccessorImpl.newInstance0(java.base@9-internal/Native Method)
     [java]     at sun.reflect.NativeConstructorAccessorImpl.newInstance(java.base@9-internal/NativeConstructorAccessorImpl.java:62)
     [java]     at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(java.base@9-internal/DelegatingConstructorAccessorImpl.java:45)
     [java]     at java.lang.reflect.Constructor.newInstance(java.base@9-internal/Constructor.java:453)
     [java]     at edu.umd.cs.findbugs.DetectorFactory$ReflectionDetectorCreator.createDetector2(DetectorFactory.java:92)
     [java]     ... 6 more
     [java] Caused by: java.lang.NoClassDefFoundError: javax/annotation/WillClose
     [java]     at edu.umd.cs.findbugs.detect.BuildObligationPolicyDatabase.<init>(BuildObligationPolicyDatabase.java:134)
     [java]     ... 11 more
     [java] Caused by: java.lang.ClassNotFoundException: javax.annotation.WillClose
     [java]     at jdk.internal.loader.BuiltinClassLoader.loadClass(java.base@9-internal/BuiltinClassLoader.java:366)
     [java]     at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(java.base@9-internal/ClassLoaders.java:184)
     [java]     at java.lang.ClassLoader.loadClass(java.base@9-internal/ClassLoader.java:419)
     [java]     ... 12 more
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Alluxio Parent ..................................... SUCCESS [  3.808 s]
[INFO] Alluxio Core ....................................... SUCCESS [  0.793 s]
[INFO] Alluxio Core - Protobuf ............................ FAILURE [  5.748 s]
[INFO] Alluxio Core - Common Utilities .................... SKIPPED
[INFO] Alluxio Core - Client .............................. SKIPPED
```

Reply:
Also, for reference, when I installed the OpenJDK 9 (`openjdk-9-jre openjdk-9-jre-headless openjdk-9-jdk openjdk-9-jdk-headless`) The result of `java -version` and `javac -version` give a version of `9-internal`

Reply:
@ZacBlanco tested with my local mac. I found Java 9 failed to build either (for different reasons).I created a JIRA https://alluxio.atlassian.net/browse/ALLUXIO-3313 to track this issue.